libs/category/src/resolvers/category-page/category-page.resolver.ts

```diff
@@ -0,0 +1,39 @@
+import { Inject, Injectable } from '@angular/core'
+import { ActivatedRouteSnapshot, Resolve } from '@angular/router'
+import { ActionsSubject, Store } from '@ngrx/store';
+import { Observable } from 'rxjs'
+import { filter, mapTo, take } from 'rxjs/operators';
+
+import { DaffCategoryActionTypes, DaffCategoryLoad, DaffCategoryLoadFailure, DaffCategoryLoadSuccess } from '../../actions/category.actions';
+import { DaffCategoryReducersState } from '../../reducers/category-reducers.interface';
+import { DaffDefaultCategoryPageSize } from './default-category-page-size.token';
+
+/**
+ * Resolves category data for category pages, and will only resolve the url after a category request succeeds or fails. This resolver expects a url
+ * of the form `some/url/{id}` where `{id}` is the category id.
+ */
+@Injectable({
+	providedIn: 'root'
+})
+export class DaffCategoryPageResolver implements Resolve<Observable<boolean>> {
+  constructor(
+		@Inject(DaffDefaultCategoryPageSize) private defaultCategoryPageSize: number,
+    private store: Store<DaffCategoryReducersState>,
+    private dispatcher: ActionsSubject,
+	) {}
+	
+	resolve(route: ActivatedRouteSnapshot): Observable<boolean> {
+		this.store.dispatch(new DaffCategoryLoad({
+      id: route.paramMap.get('id'), page_size: this.defaultCategoryPageSize
+		}));
+
+		return this.dispatcher.pipe(
+			filter((action: DaffCategoryLoadSuccess | DaffCategoryLoadFailure) => 
+				action.type === DaffCategoryActionTypes.CategoryLoadSuccessAction
+				|| action.type === DaffCategoryActionTypes.CategoryLoadFailureAction
+			),
```

Comment:
Same here. If `ofType` works, you should use it here.

Reply:
changed